dot/rpc/modules/payment.go

```diff
@@ -70,3 +84,30 @@ func (p *PaymentModule) QueryInfo(_ *http.Request, req *PaymentQueryInfoRequest,
 
 	return nil
 }
+
+// getRuntimeFromDB gets the runtime for the corresponding block hash from storageState
+func (p *PaymentModule) getRuntimeFromDB(blockHash *common.Hash) (instance runtime.Instance, err error) {
+	var stateRootHash *common.Hash
+	if blockHash != nil {
+		stateRootHash, err = p.storageAPI.GetStateRootFromBlock(blockHash)
```

Comment:
It would be nice to use a  method on storage state if that's possible, so we can dedup this code which is here twice.